vscode/src/debugger/session.ts

```diff
@@ -131,7 +167,17 @@ export class QscDebugSession extends LoggingDebugSession {
   protected async launchRequest(
     response: DebugProtocol.LaunchResponse,
     args: ILaunchRequestArguments
-  ) {
+  ): Promise<void> {
+    if (this.failed) {
+      log.trace("compilation failed. sending error response");
```

Comment:
Minor nit, but this seems more like an info level, or at least debug. I think of trace as being for super-verbose stuff to track the path of execution and relevant data - e.g. function entry/exit, details of network requests, etc.